modules/webgl/src/classes/resource.js

```diff
@@ -71,6 +72,20 @@ export default class Resource {
     return this;
   }
 
+  addRef() {
+    assert(this._refCount > 0);
+    this._refCount++;
+    return this;
+  }
+
+  removeRef() {
+    assert(this._refCount > 0);
+    this._refCount--;
+    if (this._refCount === 0) {
+      this.delete();
```

Comment:
I'm fine with this being an opt-in behavior, but `delete` should check refCount as well?

Reply:
But this will break the opt-in behavior. Old users will call delete but it won't delete because ref count is 1.

Reply:
How is ref count set if you don't call addRef?

Reply:
I initially set it to `1`. We can have delete() print a warning instead.

Reply:
This way the users of the old API/way will always get a warning.